web-client/public/gene/api.js

```diff
@@ -199,56 +209,97 @@
+                    syntheticLethality: "Not found", // Information unavailable via regular API
+                    syntheticRescue: "Not found", // Information unavailable via regular API
                     geneOntologySummary: data.functionSummary,
-                    molecularFunction: "N/A", // Information unavailable via regular API
-                    biologicalProcess: "N/A", // Information unavailable via regular API
-                    cellularComponent: "N/A", // Information unavailable via regular API
+                    molecularFunction: "Not found", // Information unavailable via regular API
+                    biologicalProcess: "Not found", // Information unavailable via regular API
+                    cellularComponent: "Not found", // Information unavailable via regular API
                 };
+
+                for (var prop in yeastmineTemplate) {
+
+                    if ((yeastmineTemplate[prop] === undefined) ||
+                    (yeastmineTemplate[prop] === null)) {
+                        yeastmineTemplate[prop] = "Not found";
+                    }
+                }
+
+
+
+                return yeastmineTemplate;
             };
 
             var parseEnsembl = function (data) {
-                return {
+
+                var ensemblTemplate = {
                     ensemblID: data.id,
                     description: data.description,
-                    dnaSequence: "N/A", // Information unavailable via regular API
-                    geneLocation: "N/A", // Information unavailable via regular API
+                    dnaSequence: "Not found", // Information unavailable via regular API
+                    geneLocation: "Not found", // Information unavailable via regular API
                 };
+
+                for (var prop in ensemblTemplate) {
+                    if ((ensemblTemplate[prop] === undefined) ||
+                    (ensemblTemplate[prop] === null)) {
+                        ensemblTemplate[prop] = "Not found";
+                    }
+                }
+                return ensemblTemplate;
+
             };
 
             var parseJaspar = function (data) {
 
-                return data ? {
+                var jasparTemplate = {
                     jasparID : data.matrix_id,
-                    class: data.class,
-                    family: data.family,
+                    class: data.class[0],
+                    family: data.family[0],
                     sequenceLogo: data.sequence_logo,
                     frequencyMatrix: data.pfm,
-                } : {};
+                };
+
+                for (var prop in jasparTemplate) {
+                    if ((jasparTemplate[prop] === undefined) ||
+                    (jasparTemplate[prop] === null)) {
+                        jasparTemplate[prop] = "Not found";
+                    }
+                }
+
+                return jasparTemplate;
```

Comment:
We're definitely going to need t capture this logic in a unit test. It will need an API-mocking tool like Sinon.js so that we can predictably return very specific responses in order to verify specific behaviors. If you can, see if you can do some advance reading on Sinon.js to get an idea for how this is done.